src/crystal/system/unix/user.cr

```diff
@@ -33,4 +36,8 @@ module Crystal::System::User
 
     from_struct(pwd) if pwd_pointer
   end
+
+  private def find_current
+    from_id?(id: LibC.getuid).not_nil!
+  end
```

Comment:
I'd change this method to be nil-able, and raise (probably in the `System::User.current`) `NotFoundError` (instead of `NilAssertionError`), like the `find_by` methods.

```suggestion
  private def find_current?
    from_id?(id: LibC.getuid)
  end
```

Reply:
Thanks, did this

Reply:
Oh sorry I'm not sure if the etiquette is to self resolve or not